iib/workers/tasks/build_merge_index_image.py

```diff
@@ -118,16 +121,26 @@ def _add_bundles_missing_in_source(
             '%s bundles have invalid version label and will be deprecated.', len(invalid_bundles)
         )
 
-    _opm_index_add(
-        base_dir,
-        missing_bundle_paths,
-        binary_image,
-        from_index=source_from_index,
-        overwrite_from_index_token=overwrite_target_index_token,
-        # Use podman until opm's default mechanism is more resilient:
-        #   https://bugzilla.redhat.com/show_bug.cgi?id=1937097
-        container_tool='podman',
-    )
+    if is_image_fbc(source_from_index):
```

Comment:
What if the target image is not FBC?

Reply:
Then the execution of the code will take the `else` branch.

Reply:
Let me try to ask a better question: What will happen if I try to merge the FBC index(source) with not FBC index (target)? I would suggest checking both the `source` and `target` indexes. Or drop a comment if we merge FBC with non-FBC the result will be "something"

Reply:
Yeah everything is all about the source index. If that one is FBC the result will be FBC as well. 
So I think you can merge FBC source and SQLite target. 

I forget about this one so than you for bringing this up. 
cc @yashvardhannanavati @chandwanitulsi 

We can discuss this behavior if it is OK or if we want to merge FBC indexes only. 
Thanks.

Reply:
The approach I would suggest is to check in the beginning if both `source` and `target` indexes are of the same type. If they're not, we simply throw an IIBError indicating the same. 

Thoughts?

Reply:
Yes, however this will not help us when we create this first FBC image, right? 
I mean we would use 4.10 as a source and 4.11 (new empty FBC image) as a target. And it would fail for us. 
Do we want it this way? What if we just check if target is FBC then the result will be FBC as well? 

Thoughts?

Reply:
+1 from me to o/p FBC if the target is FBC index.
Other option we can consider is using a flag to specify the type of resultant image

Reply:
I am fine with just checking a target image. But it would be helpful having a comment above.